casanovo/denovo/model_runner.py

```diff
@@ -116,36 +118,52 @@ def train(
             self.loaders.val_dataloader(),
         )
 
-    def evaluate(self, peak_path: Iterable[str]) -> None:
-        """Evaluate peptide sequence preditions from a trained Casanovo model.
+    def log_metrics(self, test_index: AnnotatedSpectrumIndex) -> None:
+        """Log pep_precision and aa_precision
+
+        Calculate and log peptide precision and amino acid precision
+        based off of model predictions and spectrum annotations
 
         Parameters
         ----------
-        peak_path : iterable of str
-            The path with MS data files for predicting peptide sequences.
-
-        Returns
-        -------
-        self
+        test_index : AnnotatedSpectrumIndex
+            Index containing the annotated spectra used to generate model
+            predictions
         """
-        self.initialize_trainer(train=False)
-        self.initialize_model(train=False)
-
-        test_index = self._get_index(peak_path, True, "evaluation")
-        self.initialize_data_module(test_index=test_index)
-        self.loaders.setup(stage="test", annotated=True)
+        model_output = [psm[0] for psm in self.writer.psms]
+        spectrum_annotations = [
+            test_index[i][4] for i in range(test_index.n_spectra)
+        ]
+        aa_precision, _, pep_precision = aa_match_metrics(
+            *aa_match_batch(
+                spectrum_annotations,
+                model_output,
+                depthcharge.masses.PeptideMass().masses,
+            )
+        )
 
-        self.trainer.validate(self.model, self.loaders.test_dataloader())
+        logger.info("Peptide Precision: %f", pep_precision)
```

Comment:
**nitpick:** Round to a fixed number of decimals for pretty printing. Potentially format as a percentage instead of a float.

Reply:
Unfortunately it doesn't look like logging statements support percentages natively (I ran into the same thing when implementing the end of run report logging) so I just manually multiply the values by 100 instead.

Reply:
Oh right, I already forgot about the lack of percentage support. Consistent decimal formatting is still good though. 👍